### ISSUE TITLE
feat(theming): add $color-hue option to color mixin (material 3)

### ISSUE THREAD

**Issue by MikaStark** [angular/components#29107]
### Feature Description

In [Using component color variantss](https://material.angular.io/guide/theming#using-component-color-variants) guide we see there is a way to override color variant used by specific components thanks to `$color-variant` option. It would be blasting to have also a `$color-hue` to let us customize the hue. I think it would be great this could also be set more globally in theme definition to avoid setting multiple times the same value.

It worth nothing to remember that was possible to do in material 2 theming through `default`, `text`, `lighter` and `darker` special hue.

### Use Case

If you want to use bright colors (like orange) as theme primary of tertiary color you will, in fact, get out of the box darker colors (to keep orange example, you will get some brown variations) which could be not accepted by your business team (as it alter branding).

Then, if you want to apply those exact colors on your primitive elements you have to override each CSS variable which is time consuming and annoying. `$color-hue` option would save a lot of time and open new way of customisation.

**Comment by amysorto** [comment]
There are [system level variables](https://github.com/angular/components/blob/main/src/material/core/tokens/_m3-tokens.scss#L179) you can access in your styles if enabled. You can override those system level tokens to change the styles of the different components. To use them you would need to include the following mixin: `@include mat.system-level-colors($my-theme)`.

Let me know if this doesn't solve your use case, thanks!

**Comment by MikaStark** [comment]
Hi @amysorto, thank you for your suggestion. Hélas, it doesn't suit the use case. In fact, your solution is affected too.

Maybe I badly described the point. Let me try again :

Imagine you hold an application using 2 main colors : indigo and orange. Those colors are part of the graphical chart of your company and you have to use it as background color for flat buttons. Here is what I need to do with m3 : 

```scss
// m3-theme.scss

// Note: Color palettes are generated from primary: #1d3794, tertiary: #ea7b59
$_palettes: (
  // ... generated with schematics
);


// In `styles.scss` right after declaring the theme : 

.app-tertiary-button {
  @include mat.button-color(m3-theme.$light-theme, $color-variant: "tertiary");
}
```

Naivly, I want and expect `.app-tertiary-button` class to use my brand orange. But in fact, I get some weird brown color derived from the expected color. 

Result with m3 which is bad :

![image](https://github.com/angular/components/assets/23034761/da782a06-c699-41d9-9900-6b05ba938c11)  

Result with previous m2 (I used a different button design but the point is the color) which is correct :

![image](https://github.com/angular/components/assets/23034761/0892b61a-4b78-4244-a322-475d00dc983b)

I understood m3 theming is very different from m2's one but we where able to define ourself a color hue as `default`. It would be great to have such feature with m3 theming like so : 

```scss
// Current m2 theme :

$app-primary: mat.define-palette($indigo-palette, 500, 300, 700); imagine "500" is the brand indigo
$app-accent: mat.define-palette($orange-palette, 300, 100, 400); // imagine "300" hue is the brand orange

$app-theme: mat.define-light-theme((
 color: (
   primary: $app-primary,
   accent: $app-accent,
   warn: $app-warn,
 ),
 typography: mat.define-typography-config(),
 density: 0,
));

// Expected m3 theme :

$light-theme: mat.define-theme(
  (
    color: (
      theme-type: light,
      primary: $_primary,
      tertiary: $_tertiary,
      primary-default-hue: 60, // <--- whould be nice to have
      tertiary-default-hue: 30, // <--- whould be nice to have
    ),
    typography: (
      // ...
    ),
  )
);
```

The only alternative for now, is to manually override EVERY css variables using tertiary color to use the hue we want.

**Comment by MikaStark** [comment]
@amysorto don’t you think this issue deserve to be reopened ? My product manager wasn’t happy to see it « precious » brand colors changed because of material 3
